### ISSUE TITLE
Better quotes normalisation with triple quotes

### ISSUE THREAD

**Issue by hauntsaninja** [psf/black#3196]
**Is your feature request related to a problem? Please describe.**

Black's general philosophy is to prefer double quotes, but only if doing so doesn't involve additional escaping.

In both of the following I'd argue triple single quotes are a better match:
```
""""How are you\""""

'''"It isn't fair", he said.'''
```
Blackened, this is:
```
""""How are you\""""

""""It isn't fair", he said."""
```
Triple single quotes would allow you to skip some escaping in the first, so changing that seems pretty straightforwardly inline with Black's philosophy. 

The second case doesn't have any escaping, but I'd argue that triple single quotes are still much more readable.

https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ACnAG1dAD2IimZxl1N_Wk8NUTbwX8wvAQ0hWbvuf_HUUSZ5NDaLIFSPsJ9ruQ2hSRp7uA53CiLufsQTkdk9vJC3eBKzl-_5hAsF8kap9-PHLcE1ySzxFbuIRScjjl65GOCpXKqFOd_18_A2iNm2dpJxKQAAAAAAhusx8D_Cgu0AAYkBqAEAAKDjl06xxGf7AgAAAAAEWVo=

**Describe the solution you'd like**

Taking into account whether the first and last characters are quotes when normalising triple quoted strings

**Comment by zsol** [comment]
I agree both of your examples are way more readable with single quotes, so I'm generally supportive of this change (not a shocker given my original implementation in #75 handled the first example like you describe here).
However, this explicitly goes against pep-8 and pep-257, which mandate triple quotes.

**Comment by mcsitter** [comment]
I would say the way black does it is excellent, and the single triple quotes do not significantly improve the readability, imo.

**Comment by chessset5** [comment]
But the single triple quotes does however mess with formatting and in the case of a mixture of double and single quotes, in example

`regex = '''"a"'b'''` which equates to `"a":'b` when running the formatter, it results in `""""a"'b"""` which can mess with some interpreters when the start of the string has an unexpected number of quotes.

So ideally in the case of single triple quotes followed by a double quote should be left alone.
